src/lib.rs

```diff
@@ -0,0 +1,457 @@
+use std::collections::HashSet;
+
+use anyhow::{anyhow, Result};
+use guest::prelude::*;
+use k8s_openapi::api::core::v1 as apicore;
+use kubewarden_policy_sdk::wapc_guest as guest;
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use crate::settings::Rule;
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<settings::Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn validate_envvar_with_rule(rule: &Rule, env_vars: &Vec<apicore::EnvVar>) -> Result<()> {
+    let error_message: &str;
+    let mut resource_env_var: HashSet<settings::EnvVar> = HashSet::new();
+    for envvar in env_vars {
+        resource_env_var.insert(settings::EnvVar {
+            name: envvar.name.clone(),
+            value: envvar.value.clone(),
+        });
+    }
+    match rule.operator {
+        settings::Operator::AllAreUsed | settings::Operator::NotAllAreUsed => {
```

Comment:
In my opinion it is clearer to have a dedicated case for each of the 4 cases. For example I would split `AllAreUsed` and `NotAllAreUsed`. Like it is done [here](https://github.com/kubewarden/volumeMounts-policy/blob/main/src/lib.rs#L126-L187)